### DISCUSSION TITLE
Lesson 6 - 'cast send' seems to not work, because 'cast call' returns 0 regardless of what was given in 'cast send'

### DISCUSSION THREAD

**Discussion by YourGothDaddy** [Cyfrin/foundry-full-course-cu#896] [Q&A] [answered]
I am following the course and I am doing 'cast send 0x91Ae281a09f0029Bd9edd989A6eA89B2617D6045 "store(uint256)" 250 --rpc-url $RPC_URL --private-key $PRIVATE_KEY', but then on 'cast call 0x91Ae281a09f0029Bd9edd989A6eA89B2617D6045 "retrieve()"' I get 0x all the time.

**Comment by shikhar229169** [comment, marked as answer]
Make sure that while deploying the contract, you used the command-line option ```--broadcast```, because if you don't include that then that particular contract deployment txn is not sent to the network and is not included in the block, instead only simulation takes place.

**Reply by YourGothDaddy** [reply]
Could you please write the complete command, since for some reason I can't seem to find where to exactly put `--broadcast` and have it working

**Reply by theirrationalone** [reply]
Append that just at last.

**Reply by msyuvaraj** [reply]
Hi , Im also having the same issue(getting 0x while cast call retrive( ) )  in lesson 6.
i did append the `--broadcast` commad while deploying and it was succefully deployed

this is the command i used for deploying
`$ forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY`

this is the command i used for interacting i.e to store the number
`$ cast send 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6  "store(uint256)" 145 --rpc-url $RPC_URL --private-key $PRIVATE_KEY`

and this is what i got when cast called

`$ cast call 0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6 "retrieve()"
0x`

could you please help me?
